src/vm/codeman.cpp

```diff
@@ -2121,6 +2121,9 @@ void CodeHeapRequestInfo::Init()
         m_pAllocator = m_pMD->GetLoaderAllocatorForCode();
     m_isDynamicDomain = (m_pMD != NULL) ? m_pMD->IsLCGMethod() : false;
     m_isCollectible = m_pAllocator->IsCollectible() ? true : false;
+#if defined(FEATURE_JIT_PITCHING)
+    m_isJitPitchedDomain = false;
```

Comment:
>m_isJitPitchedDomain = false; [](start = 4, length = 29)

It still seems like there is much more special casing here than is necessary. Does it cause issues to use m_DynamicDomainCodeHeaps without creating a 2nd set of heaps?
Similarly it wasn't clear that we needed a seperate m_isJitPitchedDomain field, IsJitPitchableDomain() accessor, or any of the checks in codemap.cpp that appeared to clone the behavior of the dynamic domain case?

Let me know if I missed something, but I'm imagining you could simply set m_isDynamicDomain = (the current logic) || (the condition for pitchable code). Then rely on m_isDynamicDomain to drive all the right behavior throughout the rest of codeman.cpp. You would probably need to duplicate the logic for detecting pitchable methods in EEJitManager::GetCodeHeapList, but two small edits still seems like a significant complexity improvement over the current set of edits.